src/cryptonote_core/blockchain.cpp

```diff
@@ -460,37 +459,55 @@ bool Blockchain::load_missing_blocks_into_oxen_subsystems(const std::atomic<bool
                 }
                 ons_iteration_duration += clock::now() - ons_start;
             }
-
-            if (m_sqlite_db && (block_height >= sqlite_height)) {
-                auto sqlite_start = clock::now();
-                if (!service_node_list.process_batching_rewards(blk)) {
-                    log::error(
-                            logcat,
-                            "Unable to process block for updating SQLite DB: {}",
-                            cryptonote::get_block_hash(blk));
-                    return false;
-                }
-                sqlite_iteration_duration += clock::now() - sqlite_start;
-            }
         }
     }
+    auto end = clock::now();
+
+    if (total_blocks > 0) {
+        // NOTE: Check that all subsystems ended up synchronised to the same height
```

Comment:
I wonder if these checks should either throw or reset these databases and re-call this function?  Either way, probably good to also check the heights are in sync at the beginning of this function as well (or 0 if the chain isn't to their relevant height yet I guess).  If that's checked already outside where this is called and I haven't gotten there yet, disregard.

Reply:
Yeah this function is somewhat resilient to mismatched heights when it's called. The SNL is responsible for adding the block to the SQL DB internally. When it does that it checks that they are in sync. That throws if it isn't.

The only ways I could think of for this to go out of sync was

```
    // These 2 systems can go out of sync, though generally only in exceptional
    // cases, for example:
    //
    //  If a block was successfully added to the SNL but an exception was
    //  thrown when trying to add to the SQL DB (maybe a logic bug, DB is
    //  corrupted or DB is not in a writable state).
    //
    //  The daemon is terminated in the middle of updating the SNL or SQL DB.
    //
    //  The SQL DB was deleted or SNL was reset.
```

What happens then is the SNL will throw, it'll abort whatever operation just happened. On startup, if we detect that the heights are not in synced, it'll detach to the lowest common shared height and then call this function to resync them. So in practice we should be somewhat resilient to out-of-sync DBs.

Reply:
ok cool.  That logic is scattered around enough that I thought it prudent to ask just in case.